e2e-tests/src/lib.rs

```diff
@@ -36,6 +36,7 @@ impl Counter {
         #[builder(default)]
         initial: usize,
     ) -> Self {
+        eprintln!("Non-const");
```

Comment:
Is this to prevent `clippy::missing_const_for_fn`? Do that explicitly rather than some magic eprintln without context?

Reply:
I use this `eprinln!()` to avoid adding a global  `allow`. This way if `bon` generates code that violates this lint (which it did before) then clippy will catch this.